src/features/vote/CommitPhase.tsx

```diff
@@ -145,30 +144,25 @@ const CommitPhase: FC<Props> = ({
             commitVotes(vc, fd)
               .then((tx) => {
                 // console.log("tx", tx);
-                setModalState("pending");
                 setSubmitErrorMessage("");
-
+                close();
+                reset();
                 // Need to confirm if the user submits the vote.
                 assert(tx, "Transaction did not get submitted, try again");
                 setTxHash(tx.hash);
                 if (notify) notify.hash(tx.hash);
 
                 tx.wait(1)
                   .then((conf: any) => {
-                    // Temporary, as mining is instant on local ganache.
-                    // setTimeout(() => setModalState("success"), 5000);
-                    setModalState("success");
                     refetchEncryptedVotes();
                     reset();
                   })
                   .catch((err: any) => {
                     setSubmitErrorMessage("Error with tx.");
```

Comment:
small nit here, could we add the error message. would be useful if there actually is an error by a user to know what the underlying error was when reported: `setSubmitErrorMessage('Error with tx: ' + err.message);`

Reply:
Sounds good!